### ISSUE TITLE
Seems that it does not have a great support for function 'reshape'

### ISSUE THREAD

**Issue by YuxuanDuan** [microsoft/nni#5449]
**Describe the bug**:
When I tried to SpeedUp my model, it reminds me that `WARNING: throw some args away when calling the function "reshape".` And I found that the the output dim and the input dim of the next layer does not match after pruning. Say, this is the origin model
`(OG): ModuleList(
    (0): Linear(in_features=2, out_features=64, bias=True)
    (1): Linear(in_features=64, out_features=128, bias=True)
    (2): Linear(in_features=128, out_features=256, bias=True)
    (3): Linear(in_features=256, out_features=512, bias=True)
    (4): Linear(in_features=512, out_features=256, bias=True)
    (5): Linear(in_features=256, out_features=128, bias=True)
    (6): Linear(in_features=128, out_features=64, bias=True)
    (7): Linear(in_features=64, out_features=2, bias=True)
  )`

After pruning , it goes like
`(QG): ModuleList(
    (0): Linear(in_features=2, out_features=32, bias=False)
    (1): Linear(in_features=58, out_features=64, bias=False)
    (2): Linear(in_features=105, out_features=128, bias=False)
    (3): Linear(in_features=203, out_features=256, bias=False)
    (4): Linear(in_features=398, out_features=128, bias=False)
    (5): Linear(in_features=188, out_features=64, bias=False)
    (6): Linear(in_features=80, out_features=32, bias=False)
    (7): Linear(in_features=42, out_features=2, bias=False)
  )`

As you see, the out_feature of layer 0 is 32, but the in_features of layer 1 is 58 instead of 32, which leads to mismatch of matrix multiplecation

**Environment**:
- NNI version:2.10
- Training service (local|remote|pai|aml|etc):local
- Python version:3.9
- PyTorch version:1.13.1
- Cpu or cuda version:

**Reproduce the problem**
- Code|Example:
As is attached
[trailnni.zip](https://github.com/microsoft/nni/files/10986709/trailnni.zip)

- How to reproduce:

**Comment by J-shang** [comment]
please have a try `python -m pip install --extra-index-url https://test.pypi.org/simple/ --no-cache-dir nni==3.0a0`,
then `from nni.compression.pytorch.speedup.v2 import ModelSpeedup`.

Or you could set a larger `confidence` in speedup v1 version.

**Comment by YuxuanDuan** [comment]
> please have a try `python -m pip install --extra-index-url https://test.pypi.org/simple/ --no-cache-dir nni==3.0a0`, then `from nni.compression.pytorch.speedup.v2 import ModelSpeedup`.
> 
> Or you could set a larger `confidence` in speedup v1 version.

Yes, thanks for your reply. After I upgrade nni to 3.0, it doesn't not give me any warning when calling 'reshape'. However, the out_feature of the former layer is still mismatch with the latter ones.

**Comment by J-shang** [comment]
hello @YuxuanDuan, could you show your code for us reproduce your issue?

**Comment by J-shang** [comment]
the dependency in your model is too complex to trace, so it makes a mistake, but fortunately, we can manually set the dependency in 3.0, please using the follow code. 

```
from nni.contrib.compression.pruning import L1NormPruner
from nni.compression.pytorch.speedup.v2 import ModelSpeedup

config_list = [{
    'op_names': [f'OG.{i}', f'QG.{i}', f'PG.{i}'],
    'sparse_ratio': 0.5,
    'dependency_group_id': i
} for i in range(7)]
pruner = L1NormPruner(model, config_list)
_, masks = pruner.compress()
pruner.unwrap_model()
ModelSpeedup(model, (example_inputs,), masks).speedup_model()
```
And we will refactor the dependency part in nni 3.1, then you could avoid manually set dependency group.
